jax/_src/lax/lax.py

```diff
@@ -1355,11 +1355,19 @@ def _broadcasting_shape_rule(name, *avals):
   if len({len(shape) for shape in shapes}) != 1:
     msg = '{}: arrays must have same number of dimensions, got {}.'
     raise TypeError(msg.format(name, ', '.join(map(str, map(tuple, shapes)))))
-  result_shape = _try_broadcast_shapes(shapes)
```

Comment:
I am curious if we can use the existing code and instead overload the `core.symbolic_equal_dim` to work on Tracers. It should be quite easy, create a subclass of `core.DimensionHandler` and set it in core._SPECIAL_DIMENSION_HANDLERS. If this is enough, then this line of work should be able to take advantage of all the adapting that has happened in the abstract evaluation rules and the jax.numpy layer.

Reply:
This sounds like a good idea! I may leave it for follow-up work though.